File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/SmartRefreshLayou/ViewAdapter.java
package com.dxmovie.dxbase.binding.viewadapter.SmartRefreshLayou;

import android.view.ViewTreeObserver;
import android.widget.ScrollView;

import androidx.core.widget.NestedScrollView;
import androidx.databinding.BindingAdapter;

import com.dxmovie.dxbase.binding.command.BindingCommand;
import com.scwang.smartrefresh.layout.SmartRefreshLayout;

/**
 *
 */
public final class ViewAdapter {

    @BindingAdapter({"refreshCommand"})
    public static void setRefreshing(SmartRefreshLayout smartRefreshLayout, BindingCommand bindingCommand) {
        if (smartRefreshLayout != null)
            smartRefreshLayout.setOnRefreshListener(refreshLayout -> {
                if (bindingCommand != null) {
                    bindingCommand.executeAction();
                }
            });
    }

    @BindingAdapter({"loadMoreCommand"})
    public static void setLoadMoreing(SmartRefreshLayout smartRefreshLayout, BindingCommand bindingCommand) {
        if (smartRefreshLayout != null)
            smartRefreshLayout.setOnLoadMoreListener(refreshLayout -> {
                if (bindingCommand != null) {
                    bindingCommand.executeAction();
                }
            });
    }

//    @BindingAdapter({"noMoreData"})
//    public static void noMoreData(SmartRefreshLayout smartRefreshLayout, boolean isNoMoreData) {
//        if (smartRefreshLayout != null) {
//            smartRefreshLayout.setNoMoreData(isNoMoreData);
//        }
//    }


    @BindingAdapter(value = {"enable",}, requireAll = false)
    public static void enable(SmartRefreshLayout smartRefreshLayout, boolean enable) {
        if (smartRefreshLayout != null) {
            smartRefreshLayout.setEnableRefresh(enable);
            smartRefreshLayout.setEnableLoadMore(enable);
        }
    }

    @BindingAdapter(value = {"autoLoadMore", "canLoadMore", "delayed"}, requireAll = false)
    public static void finishLoadMore(SmartRefreshLayout smartRefreshLayout, boolean autoLoadMore, boolean canLoadMore, int delayed) {

        if (autoLoadMore) {
            smartRefreshLayout.autoLoadMore();
        } else {

            if (canLoadMore) {
                smartRefreshLayout.finishLoadMore(delayed);
            } else {
                smartRefreshLayout.finishLoadMoreWithNoMoreData();
            }
        }
    }


    @BindingAdapter({"autoRefresh"})
    public static void autoRefresh(SmartRefreshLayout smartRefreshLayout, boolean isAuto) {
        if (smartRefreshLayout != null) {
            if (isAuto) {
                smartRefreshLayout.autoRefresh();
            } else {
                smartRefreshLayout.finishRefresh();
            }
        }
    }
}

File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/SmartRefreshLayou/ViewAdapter.kt
package com.dxmovie.dxbase.binding.viewadapter.SmartRefreshLayou

import androidx.databinding.BindingAdapter
import com.dxmovie.dxbase.binding.command.BindingCommand
import com.scwang.smartrefresh.layout.SmartRefreshLayout
import com.scwang.smartrefresh.layout.api.RefreshLayout

class ViewAdapter {

    companion object{
        @JvmStatic
        @BindingAdapter("refreshCommand")
        fun setRefreshing(smartRefreshLayout: SmartRefreshLayout?, bindingCommand: BindingCommand?) {
            smartRefreshLayout?.setOnRefreshListener { refreshLayout: RefreshLayout? ->
                bindingCommand?.executeAction()
            }
        }

        @JvmStatic
        @BindingAdapter("loadMoreCommand")
        fun setLoadMoreing(smartRefreshLayout: SmartRefreshLayout?, bindingCommand: BindingCommand?) {
            smartRefreshLayout?.setOnLoadMoreListener { refreshLayout: RefreshLayout? ->
                bindingCommand?.executeAction()
            }
        }

//    @BindingAdapter({"noMoreData"})
//    public static void noMoreData(SmartRefreshLayout smartRefreshLayout, boolean isNoMoreData) {
//        if (smartRefreshLayout != null) {
//            smartRefreshLayout.setNoMoreData(isNoMoreData);
//        }
//    }


        //    @BindingAdapter({"noMoreData"})
//    public static void noMoreData(SmartRefreshLayout smartRefreshLayout, boolean isNoMoreData) {
//        if (smartRefreshLayout != null) {
//            smartRefreshLayout.setNoMoreData(isNoMoreData);
//        }
//    }
        @JvmStatic
        @BindingAdapter(value = ["enable"], requireAll = false)
        fun enable(smartRefreshLayout: SmartRefreshLayout?, enable: Boolean) {
            if (smartRefreshLayout != null) {
                smartRefreshLayout.setEnableRefresh(enable)
                smartRefreshLayout.setEnableLoadMore(enable)
            }
        }

        @JvmStatic
        @BindingAdapter(value = ["autoLoadMore", "canLoadMore", "delayed"], requireAll = false)
        fun finishLoadMore(smartRefreshLayout: SmartRefreshLayout, autoLoadMore: Boolean, canLoadMore: Boolean, delayed: Int) {
            if (autoLoadMore) {
                smartRefreshLayout.autoLoadMore()
            } else {
                if (canLoadMore) {
                    smartRefreshLayout.finishLoadMore(delayed)
                } else {
                    smartRefreshLayout.finishLoadMoreWithNoMoreData()
                }
            }
        }


        @JvmStatic
        @BindingAdapter("autoRefresh")
        fun autoRefresh(smartRefreshLayout: SmartRefreshLayout?, isAuto: Boolean) {
            if (smartRefreshLayout != null) {
                if (isAuto) {
                    smartRefreshLayout.autoRefresh()
                } else {
                    smartRefreshLayout.finishRefresh()
                }
            }
        }
    }
}